### PR TITLE
[WIP] Update dependency guidelines

### PR DESCRIPTION
**Pull request by bording**
Still working on this. The goal is to revise the guidelines to keep the intent for each category, but get rid of the wildcard usage.

### CODE REVIEW

File: guidelines/dependency-management-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+##### Unstable
+
+We want downstreams to pick up the latest core alpha/beta/rc immediately to quickly catch problems.
+
+PackageReference: `[Major.Minor.0-*, Major+1)`
+
+#### Tests
+
+Examples are container tests, acceptance testing, acceptance tests, transport tests etc
+
```
Comment (andreasohlund) [inline guidelines/dependency-management-guidelines.md]: No guidance here, I assume we want to use the latest?

File: guidelines/dependency-management-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+PackageReference: `[Major.Minor.0, Major+1)`
+
+#### Tests
+
+We want to compile against the latest version to catch breaking changes and bugs. This is the likely version that our users would be on.
+
+PackageReference: `[Major.*, Major+1)`
+
+### Core
```
Comment (andreasohlund) [inline guidelines/dependency-management-guidelines.md]: Can we make the more clear? Perhaps "Downstreams referencing NServiceBus.Core" ?

File: guidelines/dependency-management-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+##### Stable
+
+We want to compile against the lowest compatible version to catch breaking changes.
+
+PackageReference: `[Major.Minor.0, Major+1)`
+
+##### Unstable
+
+We want downstreams to pick up the latest core alpha/beta/rc immediately to quickly catch problems.
```
Comment (andreasohlund) [inline guidelines/dependency-management-guidelines.md]: Perhaps clarify when downstreams are in this "mode" => when we're about to release a major? (or minors also use this procedure)

**Comment by bording** [comment]
@andreasohlund All of that will be addressed when I actually rewrite the guidance. So far that was just a copy + paste from the coding guidelines page!

**Comment by andreasohlund** [comment]
Sorry, thought you asked for a review but realized it was probably GH that
auto added me for a review 😂

**Comment by timbussmann** [comment]
still wip?

**Comment by bording** [comment]
> still wip?

@timbussmann Yes, I haven't made any progress on this yet.

File: guidelines/dependency-management-guidelines.md (hunk 3 above)
Comment (adamralph) [reply guidelines/dependency-management-guidelines.md]: I think minors should use the same procedure.

**Pull request closed without merging**